hatch/src/yaml.rs

```diff
@@ -1,140 +1,159 @@
-            .filter(|&(k, v)| k.is_some() && v.is_some())
-            .map(|(k, v)| (k.unwrap(), v.unwrap()))
-            .collect::<Vec<(&str, &str)>>();
-
-        for (key, value) in items {
-            match key {
-                "kind" => {
-                    kind = match value {
-                        "static-lib" => ProjectKind::Library(LibraryKind::Static),
-                        "shared-lib" => ProjectKind::Library(LibraryKind::Shared),
-                        _ => ProjectKind::Binary,
-                    }
-                }
-                "compiler" => {
-                    compiler = String::from(value);
-                }
-                "compiler_flags" => {
-                    compiler_flags = String::from(value)
-                        .split_whitespace()
-                        .map(String::from)
-                        .collect();
-                }
-                "linker_flags" => {
-                    linker_flags = String::from(value)
-                        .split_whitespace()
-                        .map(String::from)
-                        .collect();
-                }
-                "arch" => {
-                    match value {
-                        "x64" => arch = Arch::X64,
-                        "x86" => arch = Arch::X86,
-                        _ => {
-                            // use default
-                        }
-                    }
-                }
-                "target" => {
-                    match value {
-                        "debug" => target = Target::Debug,
-                        "release" => target = Target::Release,
-                        _ => {
-                            // use default
-                        }
-                    }
-                }
-                _ => {}
-            }
-        }
-    } else {
-        return Err(MissingBuildError)?;
-    }
-
-    if let Some(d) = yml_vec[0]["deps"].as_hash() {
-        deps = d.iter()
-            .filter_map(|(_k, v)| v.as_str())
-            .map(|v| Dependency::new(v.to_owned()))
-            .collect();
-    } else {
-        deps = Vec::new();
-    }
-
-    let config = Config::new(kind, compiler, compiler_flags, linker_flags, arch, target);
-    Ok(Project::new(
-        name,
-        version,
-        config,
-        deps,
-        PathBuf::from(path),
-    ))
-}
-
-fn from_file(file_name: &Path) -> HatchResult<Vec<Yaml>> {
-    let mut file = fs::File::open(&file_name)
-        .with_context(|_| format!("failed to open file: `{}`", &file_name.display()))?;
-
-    let mut contents = String::new();
-    file.read_to_string(&mut contents)
-        .with_context(|_| format!("failed to read contents of: `{}`", file_name.display()))?;
-
-    let res = YamlLoader::load_from_str(&contents)
-        .compat()
-        .with_context(|e| format!("Parsing error: `{}`", e))?;
-
-    Ok(res)
-}
-
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {}
-}
+//use std::fs;
```

Comment:
We no longer need this 150 line file cause serde_yaml

Reply:
🥇